netann/chan_status_manager_test.go

```diff
@@ -0,0 +1,648 @@
+		numChannels:        numChannels,
+		graph:              graph,
+		htlcSwitch:         htlcSwitch,
+		mgr:                mgr,
+		safeDisableTimeout: cfg.ChanInactiveTimeout + time.Second,
+	}
+
+	// Initialize link status as requested.
+	if startActive {
+		h.markActive(h.graph.channels)
+	} else {
+		h.markInactive(h.graph.channels)
+	}
+
+	return h
+}
+
+// markActive updates the active status of the passed channels within the mock
+// switch to active.
+func (h *testHarness) markActive(channels []*channeldb.OpenChannel) {
+	h.t.Helper()
+
+	for _, channel := range channels {
+		chanID := lnwire.NewChanIDFromOutPoint(&channel.FundingOutpoint)
+		h.htlcSwitch.SetStatus(chanID, true)
+	}
+}
+
+// markInactive updates the active status of the passed channels within the mock
+// switch to inactive.
+func (h *testHarness) markInactive(channels []*channeldb.OpenChannel) {
+	h.t.Helper()
+
+	for _, channel := range channels {
+		chanID := lnwire.NewChanIDFromOutPoint(&channel.FundingOutpoint)
+		h.htlcSwitch.SetStatus(chanID, false)
+	}
+}
+
+// assertEnables requests enables for all of the passed channels, and asserts
+// that the errors returned from RequestEnable matches expErr.
+func (h *testHarness) assertEnables(channels []*channeldb.OpenChannel, expErr error) {
+	h.t.Helper()
+
+	for _, channel := range channels {
+		h.assertEnable(channel.FundingOutpoint, expErr)
+	}
+}
+
+// assertDisables requests disables for all of the passed channels, and asserts
+// that the errors returned from RequestDisable matches expErr.
+func (h *testHarness) assertDisables(channels []*channeldb.OpenChannel, expErr error) {
+	h.t.Helper()
+
+	for _, channel := range channels {
+		h.assertDisable(channel.FundingOutpoint, expErr)
+	}
+}
+
+// assertEnable requests an enable for the given outpoint, and asserts that the
+// returned error matches expErr.
+func (h *testHarness) assertEnable(outpoint wire.OutPoint, expErr error) {
+	h.t.Helper()
+
+	err := h.mgr.RequestEnable(outpoint)
+	if err != expErr {
+		h.t.Fatalf("expected enable error: %v, got %v", expErr, err)
+	}
+}
+
+// assertDisable requests a disable for the given outpoint, and asserts that the
+// returned error matches expErr.
+func (h *testHarness) assertDisable(outpoint wire.OutPoint, expErr error) {
+	h.t.Helper()
+
+	err := h.mgr.RequestDisable(outpoint)
+	if err != expErr {
+		h.t.Fatalf("expected disable error: %v, got %v", expErr, err)
+	}
+}
+
+// assertNoUpdates waits for the specified duration, and asserts that no updates
+// are announced on the network.
+func (h *testHarness) assertNoUpdates(duration time.Duration) {
+	h.t.Helper()
+
+	h.assertNumUpdates(0, false, duration)
+}
+
+// assertNumUpdates waits for the specified duration, asserting that n updates
+// are received on the network and that all of their disable bits are set to
+// match expEnabled. The expEnabled parameter is ignored if n is 0.
+func (h *testHarness) assertNumUpdates(n int, expEnabled bool, duration time.Duration) {
```

Comment:
Could rework this so that we also take in the channels which we expect to have new updates? This way we can ensure there's exactly one update per channel.